kensu/sdk/__init__.py

```diff
@@ -409,7 +409,23 @@ def get_datasources(self):
         return self.requests_get_json("/api/services/v1/resources/datasources")
 
     def get_logical_datasources(self):
-        return self.requests_get_json("/business/services/views/v1/logical-datasources")
+        logical_data_sources = {'data': []}
+        limit = 500
+        offset = 0
+
+        ds = {'data': [None]} 
+    
+        while ds['data']:
+            try:
+                ds = self.requests_get_json(f"/business/services/views/v1/logical-datasources?offset={offset}&limit={limit}")
+                logical_data_sources['data'].extend(ds['data'])
+                offset += limit
+                
+            except Exception as e:
+                print(f"Error fetching data: {e}")
+                break # Exit loop on error
```

Comment:
good that we have `break` here. 
@SammyEK if we didn't have it,  in case of repeated http exception in `self.requests_get_json` after some successful fetch(es), the `while` would loop forever because `ds` var won't get updated